File: src/app/utils/validation.ts
export const isValidEmail = (email: string) =>
  /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email.trim());

export const validatePassword = (password: string) => {
  if (password.length < 8) return "Password must be at least 8 characters";
  return "";
};

File: backend/utils/validation.py
import re


EMAIL_PATTERN = re.compile(r"^[^\s@]+@[^\s@]+\.[^\s@]+$")


def validate_email(email: object) -> str:
    if not isinstance(email, str) or not EMAIL_PATTERN.fullmatch(email.strip()):
        return "Please enter a valid email address"
    return ""


def validate_password(password: object) -> str:
    if not isinstance(password, str) or len(password) < 8:
        return "Password must be at least 8 characters"
    return ""
